### PR TITLE
feat: Office Protect process

### PR DESCRIPTION
**Pull request by delepster**
## Summary of changes

- added templates
- added code owners
- added azdo pipelines
- applied code cleanup
- updated package generation
- kept only net6 target framework
- remove demo projects

Related work item: [AB#37280](https://dev.azure.com/SWEngineering/8e6bc3ba-ba84-4595-8560-f51b3fe8e2b5/_workitems/edit/37280)

### CODE REVIEW

File: Mongo.Migration.Test/Documents/Serializers/DocumentVersionSerializer_when_serialize_and_deserialize.cs [hunk 1]
```diff
@@ -63,16 +38,43 @@ public void Then_version_is_serialized_correct()
...
             // Act 
-            _serializer.Serialize(context, args, version);
+            this._serializer.Serialize(context, args, version);
 
             // Assert 
             BsonDocument document = writer.Document;
             document.ToString().Should().Be("{ \"version\" : \"0.0.1\" }");
         }
+
+        #region SetUp
```
Comment (char1es31) [inline Mongo.Migration.Test/Documents/Serializers/DocumentVersionSerializer_when_serialize_and_deserialize.cs:52]: you decide to keep the regions? is this intended, I thought our team didn't like them

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -0,0 +1,2 @@
+# Make the Turing team owner of the entire repo
+* @sherweb/turing
```
Comment (guyglies100) [inline .github/CODEOWNERS:2]: 🙏

File: Mongo.Migration.Test/Documents/Serializers/DocumentVersionSerializer_when_serialize_and_deserialize.cs (hunk 1 above)
Comment (guyglies100) [inline Mongo.Migration.Test/Documents/Serializers/DocumentVersionSerializer_when_serialize_and_deserialize.cs]: ❌Remove pragma regions

**Review by char1es31 — APPROVED** [review]

File: Mongo.Migration.Test/Performance/PerformanceOnStartup.cs [hunk 3]
```diff
@@ -19,15 +23,52 @@ public class PerformanceTestOnStartup
...
+            var result = swWithMigration.ElapsedMilliseconds - sw.ElapsedMilliseconds;
+
+            Console.WriteLine(
+                $"MongoDB: {sw.ElapsedMilliseconds}ms, Mongo.Migration: {swWithMigration.ElapsedMilliseconds}ms, Diff: {result}ms (Tolerance: {TOLERANCE_MS}ms), Documents: {DOCUMENT_COUNT}, Migrations per Document: 2");
+
+            // Assert
+            result.Should().BeLessThan(TOLERANCE_MS);
         }
 
         #region PRIVATE
```
Comment (guyglies100) [inline Mongo.Migration.Test/Performance/PerformanceOnStartup.cs]: remove pragma regions

File: Mongo.Migration.Test/Performance/PerformanceOnStartup.cs [hunk 4]
```diff
@@ -50,80 +92,46 @@ private void InsertMany(int number, bool withVersion)
...
+            this.MigrateAll(false);
         }
 
         private void ClearCollection()
         {
-            _client.GetDatabase(DATABASE_NAME).DropCollection(COLLECTION_NAME);
+            this._client.GetDatabase(DATABASE_NAME).DropCollection(COLLECTION_NAME);
         }
 
         #endregion
```
Comment (guyglies100) [inline Mongo.Migration.Test/Performance/PerformanceOnStartup.cs]: remove pragma regions

File: Mongo.Migration/Migrations/Adapters/LightInjectAdapter.cs [hunk 5]
```diff
@@ -9,34 +10,36 @@ public class LightInjectAdapter : IContainerAdapter
 
         public LightInjectAdapter(IServiceContainer container)
```
Comment (guyglies100) [inline Mongo.Migration/Migrations/Adapters/LightInjectAdapter.cs:11]: LightInjectAdapter I would maybe add autofac adapter and or remove this

File: Mongo.Migration/Migrations/Locators/TypeMigrationDependencyLocator.cs [hunk 6]
```diff
@@ -2,61 +2,62 @@
...
         {
             var migrationTypes =
-                (from assembly in Assemblies
-                from type in assembly.GetTypes()
-                where typeof(TMigrationType).IsAssignableFrom(type) && !type.IsAbstract
-                select type).Distinct(new TypeComparer());
+                (from assembly in this.Assemblies
+                 from type in assembly.GetTypes()
+                 where typeof(TMigrationType).IsAssignableFrom(type) && !type.IsAbstract
+                 select type).Distinct(new TypeComparer());
```
Comment (guyglies100) [inline Mongo.Migration/Migrations/Locators/TypeMigrationDependencyLocator.cs:27]: I hate this syntax

File: Mongo.Migration/Migrations/Locators/TypeMigrationLocator.cs [hunk 7]
```diff
@@ -10,12 +11,12 @@ internal class TypeMigrationLocator : MigrationLocator<IDocumentMigration>
...
         {
             var migrationTypes =
-                (from assembly in Assemblies
-                from type in assembly.GetTypes()
-                where typeof(IDocumentMigration).IsAssignableFrom(type) && !type.IsAbstract
-                select type).Distinct();
+                (from assembly in this.Assemblies
+                 from type in assembly.GetTypes()
+                 where typeof(IDocumentMigration).IsAssignableFrom(type) && !type.IsAbstract
+                 select type).Distinct();
```
Comment (guyglies100) [inline Mongo.Migration/Migrations/Locators/TypeMigrationLocator.cs:17]: I hate this syntax

File: Mongo.Migration/Migrations/Locators/TypeMigrationLocator.cs [hunk 8]
```diff
@@ -10,12 +11,12 @@ internal class TypeMigrationLocator : MigrationLocator<IDocumentMigration>
...
 
-            Migrations = migrationTypes.Select(t => (IDocumentMigration) Activator.CreateInstance(t)).ToMigrationDictionary();
+            this.Migrations = migrationTypes.Select(t => (IDocumentMigration)Activator.CreateInstance(t)).ToMigrationDictionary();
```
Comment (guyglies100) [inline Mongo.Migration/Migrations/Locators/TypeMigrationLocator.cs:19]: soft cast, I think you may une a templated method here

File: Mongo.Migration/Startup/Static/ComponentRegistry.cs [hunk 9]
```diff
@@ -7,65 +8,70 @@
...
-            _containerAdapter.Register<IMigrationService, MigrationService>();
+            this._containerAdapter.Register<IMigrationService, MigrationService>();
         }
 
-        public TComponent Get<TComponent>() where TComponent : class
+        public TComponent Get<TComponent>()
+            where TComponent : class
         {
-            return (TComponent) _containerAdapter.GetInstance(typeof(TComponent));
+            return (TComponent)this._containerAdapter.GetInstance(typeof(TComponent));
```
Comment (guyglies100) [inline Mongo.Migration/Startup/Static/ComponentRegistry.cs:46]:
```suggestion
            return this._containerAdapter.GetInstance<TComponent>();
```

**Review by guyglies100 — APPROVED** [review]
Small comments

**Pull request merged**